dataprofiler/profilers/profile_builder.py

```diff
@@ -704,14 +704,19 @@ def tqdm(l):
                     options=structured_options
                 )
                 new_cols.add(col)
-                        
-        # Format the data
-        notification_str = "Finding the Null values in the columns..."
+                
+        # Generate pool and estimate datasize
         pool = None
         if options.structured_options.multiprocess.is_enabled:
```

Comment:
my concern with setting this here rather than determined in options is that if a user sets True in MP, it will force it to False.
Maybe default in options should be `None` and if `None` then it determines here or options does auto determination if is `None`

Reply:
This really doesn't change the functionality? I just moved something that was previously out of the for loop into it?

I agree it might need to move, but isn't that a seperate PR as it's changing functionality of options (as opposed to protecting threading)?

Reply:
I would say the data size check is new functionality, this would potentially change the outcome of options as opposed to right now where a user sets the option and it uses it.

Reply:
I suggested an update to the readme:

> **multiprocess** - Option to enable multiprocessing. Automatically selects the optimal number of processes to utilize based on the dataset and system constraints.

I think this covers the concern. Personally, I think we need to modify this to allow users to set their pool size. In which case, then I'd suggest we utilize what they specified and if they say `None` then we utilize the proposed method.

Reply:
Going to move modifications to the pool based on dataset size to separate PR, which would also need to include user defined process counts